packages/astro/test/astro-dev-headers.test.js

```diff
@@ -31,10 +31,10 @@ describe('Astro dev headers', () => {
 			assert.equal(Object.fromEntries(result.headers)['x-astro'], headers['x-astro']);
 		});
 
-		it('does not return custom headers for invalid URLs', async () => {
+		it('returns custom headers in the default 404 response', async () => {
 			const result = await fixture.fetch('/bad-url');
 			assert.equal(result.status, 404);
-			assert.equal(Object.fromEntries(result.headers).hasOwnProperty('x-astro'), false);
+			assert.equal(Object.fromEntries(result.headers).hasOwnProperty('x-astro'), true);
 		});
```

Comment:
The test checks that configured headers added in `config.server.headers` are NOT added in the 404 responses.

This behavior was limited to astro's default 404 page - if the user added a custom `404.astro`, `server.headers` are still added.

The new behavior is that `server.headers` apply to all astro-generated responses - the distinction between astro's default 404 page and user's `404.astro` is gone.